src/chainstate/stacks/boot/pox_2_tests.rs

```diff
@@ -0,0 +1,322 @@
+                    alice_account.stx_balance.amount_unlocked(),
+                    1024 * POX_THRESHOLD_STEPS_USTX
+                );
+                assert_eq!(alice_account.stx_balance.amount_locked(), 0);
+                assert_eq!(alice_account.stx_balance.unlock_height(), 0);
+            }
+            let min_ustx = with_sortdb(&mut peer, |ref mut chainstate, ref sortdb| {
+                chainstate.get_stacking_minimum(sortdb, &tip_index_block)
+            })
+            .unwrap();
+            assert_eq!(min_ustx, total_liquid_ustx / 480);
+
+            // no reward addresses
+            let reward_addrs = with_sortdb(&mut peer, |ref mut chainstate, ref sortdb| {
+                get_reward_addresses_with_par_tip(chainstate, &burnchain, sortdb, &tip_index_block)
+            })
+            .unwrap();
+            assert_eq!(reward_addrs.len(), 0);
+
+            // record the first reward cycle when Alice's tokens get stacked
+            let tip_burn_block_height =
+                get_par_burn_block_height(peer.chainstate(), &tip_index_block);
+            alice_reward_cycle = 1 + burnchain
+                .block_height_to_reward_cycle(tip_burn_block_height)
+                .unwrap() as u128;
+            let cur_reward_cycle = burnchain
+                .block_height_to_reward_cycle(tip_burn_block_height)
+                .unwrap() as u128;
+
+            eprintln!(
+                "\nalice reward cycle: {}\ncur reward cycle: {}\n",
+                alice_reward_cycle, cur_reward_cycle
+            );
+        } else {
+            // Alice's address is locked as of the next reward cycle
+            let tip_burn_block_height =
+                get_par_burn_block_height(peer.chainstate(), &tip_index_block);
+            let cur_reward_cycle = burnchain
+                .block_height_to_reward_cycle(tip_burn_block_height)
+                .unwrap() as u128;
+
+            let (min_ustx, reward_addrs, total_stacked) =
+                with_sortdb(&mut peer, |ref mut c, ref sortdb| {
+                    (
+                        c.get_stacking_minimum(sortdb, &tip_index_block).unwrap(),
+                        get_reward_addresses_with_par_tip(c, &burnchain, sortdb, &tip_index_block)
+                            .unwrap(),
+                        c.test_get_total_ustx_stacked(sortdb, &tip_index_block, cur_reward_cycle)
+                            .unwrap(),
+                    )
+                });
+
+            eprintln!("\ntenure: {}\nreward cycle: {}\nmin-uSTX: {}\naddrs: {:?}\ntotal_liquid_ustx: {}\ntotal-stacked: {}\n", tenure_id, cur_reward_cycle, min_ustx, &reward_addrs, total_liquid_ustx, total_stacked);
+
+            if cur_reward_cycle >= alice_reward_cycle {
+                if cur_reward_cycle < first_v2_cycle as u128 {
+                    let (amount_ustx, pox_addr, lock_period, first_reward_cycle) =
+                        get_stacker_info(&mut peer, &key_to_stacks_addr(&alice).into()).unwrap();
```

Comment:
Does `get_stacker_info()` know to read `.pox-2` if the height is in the 2.1 epoch?

Reply:
No, it doesn't -- that function always reads data from `.pox` directly.